activerecord/lib/active_record/model_schema.rb

```diff
@@ -513,7 +512,7 @@ def compute_table_name
             "#{full_table_name_prefix}#{contained}#{undecorated_table_name(name)}#{full_table_name_suffix}"
           else
             # STI subclasses always use their superclass' table.
-            base.table_name
+            base_class.table_name
```

Comment:
Is this a performance hot spot? Assuming the local variable is there for a reason 😅

Reply:
I think not: the result of this method is cached inside `@table_name` and L512 with its string construction is way slower than the `else` case.